server/auth/auth.go

```diff
@@ -326,9 +332,15 @@ func (auth *AuthManager) Penalize(user account.AccountID, rule account.Rule) { /
 		log.Errorf("no client to penalize")
 		return
 	}
+	if auth.anarchy {
+		log.Infof("user %v penalized for rule %v, but not enforcing it", user, rule)
+		return
+	}
 	auth.storage.CloseAccount(client.acct.ID, rule)
-	client.conn.Banish() // May not want to do this. Leaving it for now.
-	auth.removeClient(client)
```

Comment:
I noticed that this was the only `removeClient`, so on failed Send and Requests, I now call `removeClient`.  However, we may wish to add the `On` method to the Link interface so that it is ultimately up to the WSLink if the connection is really dead.  On the other hand, auth'd clients could stay in the AuthManager's clients map forever, but I don't like having it grow without bounds.